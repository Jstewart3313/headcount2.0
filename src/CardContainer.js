import React from 'react'
import Card from './Card'
import PropTypes from 'prop-types'


const CardContainer = ( {data} ) => {
  const dataKeys = Object.keys(data)
  let counter = 0
  const exp = dataKeys.map( point =>
    <Card {...data[point]} className={'card' + [point]} key={counter++} />
  )
  return (
        <div className='card-container'>
         { exp }
        </div>
  )
}



CardContainer.propTypes = {
  data: PropTypes.object.isRequired
}

export default CardContainer