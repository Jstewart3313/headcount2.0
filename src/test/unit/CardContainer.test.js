import React from 'react'
import { mount, shallow } from 'enzyme'

import CardContainer from '../../CardContainer'

describe('CardContainer', () => {

  it('should match the snapshot with no thoughts', () => {
    const mockData = {
      location: 'colorado',
      stats: {
        '2017': .017
      }
    }
    const wrapper = shallow(<CardContainer data={mockData} />);
    expect(wrapper).toMatchSnapshot()
  });

})