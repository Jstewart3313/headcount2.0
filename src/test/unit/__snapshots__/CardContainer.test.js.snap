// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot with no thoughts 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    data={
        Object {
            "location": "colorado",
            "stats": Object {
              "2017": 0.017,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          0="c"
          1="o"
          2="l"
          3="o"
          4="r"
          5="a"
          6="d"
          7="o"
/>,
        <Card
          2017={0.017}
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "0": "c",
          "1": "o",
          "2": "l",
          "3": "o",
          "4": "r",
          "5": "a",
          "6": "d",
          "7": "o",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "2017": 0.017,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            0="c"
            1="o"
            2="l"
            3="o"
            4="r"
            5="a"
            6="d"
            7="o"
/>,
          <Card
            2017={0.017}
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "0": "c",
            "1": "o",
            "2": "l",
            "3": "o",
            "4": "r",
            "5": "a",
            "6": "d",
            "7": "o",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "2017": 0.017,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
